static/app/components/events/interfaces/spans/utils.tsx

```diff
@@ -514,15 +518,18 @@ function hasFailedThreshold(marks: Measurements): boolean {
   );
 
   return records.some(record => {
-    const value = marks[record.slug];
+    const {value} = marks[record.slug];
     if (typeof value === 'number' && typeof record.poorThreshold === 'number') {
       return value >= record.poorThreshold;
     }
     return false;
   });
 }
 
-export function getMeasurements(event: EventTransaction): Map<number, VerticalMark> {
+export function getMeasurements(
+  event: EventTransaction,
+  generateBounds: (bounds: SpanBoundsType) => SpanGeneratedBoundsType
+): Map<number, VerticalMark> {
```

Comment:
This function previously used to merge measurements only if they had the exact same timestamp. I've changed this so that it uses `generateBounds` on the timestamp of the measurement to get its position (in the range of [0, 1]), and then converts it to a percentage and rounds it. It then compares it to all other measurements, and if they are close enough (within 10% of each other), then they get merged.